src/as-release.c

```diff
@@ -880,6 +989,10 @@ as_release_load_from_xml (AsRelease *release, AsContext *ctx, xmlNode *node, GEr
 	xmlNode *iter;
 	gchar *prop;
 
+	g_return_val_if_fail (AS_IS_RELEASE (release), FALSE);
+	g_return_val_if_fail (AS_IS_CONTEXT (ctx), FALSE);
+	g_return_val_if_fail (node != NULL, FALSE);
```

Comment:
Not sure if keeping these checks here makes sense - this function as well as `as_release_to_xml_node` and `as_release_load_from_yaml` and `as_release_emit_yaml` are internal functions that are called thousands of times in many occasions and have all their input verified already.
Then again, the extra checks will most likely not slow these functions down by any noticeable amount (but if any of these checks would ever trigger it would be a fatal error anyway).

Reply:
Indeed, these are internal function, one possibility would be to have these as assert to make sure that nothing is calling them with wrong arguments in the future but that's not for this MR